Task6/AbstractFactory/EbookLibrayAbstractFactory.cs

```diff
@@ -0,0 +1,166 @@
+﻿using Microsoft.VisualBasic.FileIO;
+using System.Text.RegularExpressions;
+
+namespace Task6.AbstractFactory
+{
+    internal class EbookLibrayAbstractFactory : ILibrayAbstractFactory
+    {
+        List<string> _uniqueFormats;
+       
+        static string GetNumberPages(string htmlContent)
+        {
+            string htmlOpenTag = $"<span itemprop=\"numberOfPages\">";
+            string htmlCloseTag = $"</span>";
+
+            int index = htmlContent.IndexOf(htmlOpenTag, StringComparison.OrdinalIgnoreCase);
+
+            if (index != -1)
+            {
+                int endIndex = htmlContent.IndexOf(htmlCloseTag, index, StringComparison.OrdinalIgnoreCase);
+
+                if (endIndex != 1)
+                {
+                    string extractedContent = htmlContent.Substring(index + htmlOpenTag.Length, endIndex - (index + htmlOpenTag.Length));
+                    return extractedContent;
+                }
+                else
+                {
+                    throw new Exception("Close Tag not founded");
+                }
+            }
+            else
+            {
+                return "NOT FOUND";
+            }
+        }
+
```

Comment:
Move DownloadHtmlAsync and  GetNumberPages methods to a separate entity (class). And use it as post-processing activity for a catalog. EbookLibrayAbstractFactory has to be unchanged, it is supposed that the existing catalog of eBooks will be processed

Reply:
Modified